spec/filters/syslog_pri_spec.rb

```diff
@@ -5,109 +5,173 @@
+          end
+        end
+
+        it "default syslog_facility is user-level" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][facility][name]")).to eq("security/authorization")
+          else
+            expect(event.get("syslog_facility")).to eq("security/authorization")
+          end
+        end
+
       end
-    end
 
-    context "when an alert messages arrive" do
-      let(:syslog_pri) { 137 }
+      context "when notice local messages arrive" do
+        let(:syslog_pri) { 165 }
+
+        it "syslog severity is notice" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][severity][name]")).to eq("notice")
+          else
+            expect(event.get("syslog_severity")).to eq("notice")
+          end
+        end
+
+        it "default syslog_facility is user-level" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][facility][name]")).to eq("local4")
+          else
+            expect(event.get("syslog_facility")).to eq("local4")
+          end
+        end
+      end
 
-      it "syslog severity is notice" do
-        subject.filter(event)
-        expect(event.get("syslog_severity")).to eq("alert")
+      context "when a debug messages arrive" do
+        let(:syslog_pri) { 191 }
+
+        it "syslog severity is notice" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][severity][name]")).to eq("debug")
+          else
+            expect(event.get("syslog_severity")).to eq("debug")
+          end
+        end
+
+        it "default syslog_facility is user-level" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][facility][name]")).to eq("local7")
+          else
+            expect(event.get("syslog_facility")).to eq("local7")
+          end
+        end
       end
 
-      it "default syslog_facility is user-level" do
-        subject.filter(event)
-        expect(event.get("syslog_facility")).to eq("local1")
+      context "when an alert messages arrive" do
+        let(:syslog_pri) { '137' }
+
+        it "syslog severity is notice" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][severity][name]")).to eq("alert")
+          else
+            expect(event.get("syslog_severity")).to eq("alert")
+          end
+        end
+
+        it "default syslog_facility is user-level" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][facility][name]")).to eq("local1")
+            expect(event.get("[log][syslog][facility][code]")).to eq(17)
+          else
+            expect(event.get("syslog_facility")).to eq("local1")
+            expect(event.get("syslog_facility_code")).to eq(17)
```

Comment:
can you use `ecs_select` to simplify the condition?
```suggestion
          expect( event.get ecs_select[disabled: "syslog_facility", v1: "[log][syslog][facility][name]"] ). to eq("local1")
          expect( event.get ecs_select[disabled: "syslog_facility_code", v1: "[log][syslog][facility][code]"] ). to eq("17")
```

Reply:
Thanks Kaise, we could do smt similar I feel like there's too much "meta-fu" in the specs already.
So I prefer to keep the ecs vs legacy asserts explicit, also `ecs_select` is an internal method to the plugin (we would need to use something else).